Fixes #15832

comma opened the pull request:
3 hours of debug.
INBUILT LOGIN DOES TWO THINGS - SETS STATOBJ OF CLIENT TO MOB, AND POPS MOB OUT OF NULLSPACE TO 1,1,1, ZERO FUCKS GIVEN. YOU WANTED TO MOVE IT LATER? FUCK YOU. YOU WANTED TO KEEP IT IN NULLSPACE FOR SOME REASON? FUCK. YOU. SILENTLY, STEALTHILY, MOB WILL ALWAYS END UP IN SPACE WHEN IT IS NOT EXPECTING IT.
Now statobj is set in code, and other thing doesn't happen.

OKAY SO THAT BROKE OTHER THINGS.

New way is to move mob to latejoin before client is put into it. Latejoin remains virtually unchanged, and on roundstart job controller still moves mob to job start locs later, so no visible difference there.

Techhead0 reviewed (dismissed):
I'm sure there's no undocumented third built-in feature of Login() lying in wait in.

PsiOmegaDelta commented:
Why are you breaking the ..() call-chain? There are other mob/login() overrides.

comma commented:
>POPS MOB OUT OF NULLSPACE TO 1,1,1, ZERO FUCKS GIVEN.

Only thing lying upper is inbuilt Login, and docs claim it does two things, one of which is unwanted, and other one is line of code I added. Other way would be a refactor to get a chunk of job code into new player code and I'd rather try simplier solution.

PsiOmegaDelta commented:
Well..
https://github.com/Baystation12/Baystation12/blob/dev/code/datums/observation/logged_in.dm#L19
https://github.com/Baystation12/Baystation12/blob/dev/code/modules/spells/spells.dm#L20
https://github.com/Baystation12/Baystation12/blob/dev/code/_onclick/hud/ability_screen_objects.dm#L159

While not 'higher' they are of 'equal' level, and not calling ..() does risk them not being called.

comma commented:
u g h

comma committed:
Fixes #15832

Another way. Spawns mob in one of latejoin spawnpoints, and lets job controller move it later if it's roundstart spawn. Not any more noticeable than current 'spawn from space', and without flooring now. Latejoin is virtually unchanged.

comma commented:
Decided in the end to actually pass spawn turf to new of mob, instead of forcemoving them later, would indeed remove need for wheelchair fixery

PsiOmegaDelta commented:
Fixes #15832.
